tools/README.md

```diff
@@ -83,3 +83,51 @@ python3 /graphstorm/tools/partition_graph.py --dataset ogbn-arxiv \
                                              --output /tmp/ogbn_arxiv_nc_train_val_1p_4t  \
                                              --is-homo     
 ```
+
+## Convert node features from distDGL format WholeGraph format
+
+Use the `convert_feat_to_wholegraph.py` script with `--dataset-path` pointing to the distDGL folder of partitions and `--feat-names` to the features you want to convert (e.g. '`paper/feat,author/feat,institution/feat`')
+```
+python3 convert_feat_to_wholegraph.py --dataset-path ogbn-mag240m-2p --feat-names paper/feat
+```
+
+The script will create a new folder '`wholegraph`' under '`ogbn-mag240m-2p`' containing the WholeGraph input files and will trim the distDGL file `node_feat.dgl` in each partition and save a backup `node_feat.dgl.bak`.
+
+The features in those files can be loaded in memory via the WholeGraph API by mentioning the folder path and feature prefix (`<node_type>~<feat_name>`).
+Below is an standalone example showing how to load the data:
+```python
+import json
+import os
+import torch
+import pylibwholegraph.binding.wholememory_binding as wmb
+import pylibwholegraph.torch as wgth
+
+with open('ogbn-mag240m/wholegraph/metadata.json') as f:
+    metadata = json.load(f)
+
+torch.cuda.set_device(int(os.environ['LOCAL_RANK']))
+wmb.init(0)
+torch.distributed.init_process_group('nccl')
+global_comm = wgth.comm.get_global_communicator()
+
+cache_policy = wgth.create_builtin_cache_policy(
+    "none", # cache type
+    "distributed",
+    "cpu",
+    "readonly", # access type
+    0.0, # cache ratio
+)
+
+paper_feat_wg = wgth.create_embedding(
+                    global_comm,
+                    'distributed',
+                    'cpu',
+                    getattr(torch, metadata['paper/feat']['dtype'].split('.')[1]),
+                    metadata['paper/feat']['shape'],
+                    optimizer=None,
+                    cache_policy=cache_policy,
+                )
+# 'part_count' is the number of partition files. For distDGL it will always be the number of machines.
+paper_feat_wg.get_embedding_tensor().from_file_prefix('ogbn-mag240m/wholegraph/paper~feat', part_count=4)
```

Comment:
TBH, I don't know why we need to split wholegraph features into 4 different files, and save all under one folder `ogbn-mag240m/wholegraph`... It would not save any space, right?

Reply:
No it does not save space. The dataset is partitioned in 4 parts. Are you suggesting to put them in separate folder? Originally, the feats will be in 
```
ogbn-mag240m/part0/node_feat.dgl 
ogbn-mag240m/part1/node_feat.dgl
...
```
now the the features are in 
```
ogbn-mag240m/wholegraph/paper~feat_part_0_of_4
ogbn-mag240m/wholegraph/paper~feat_part_1_of_4
...
```

Reply:
Yes, saving them in separate folders is better as it allows parallel data distribution in clusters.

Reply:
That's true! But I wonder if it has something to do with WholeGraph dataloading. If it expects all the files to be in the same folder where different nodes reads parts of it? 

The file naming convention requires you to provide `num_parts` which outputs `_part_0_of_`
```
filename = wgth.utils.get_part_file_name(feat.replace('/', '~'), part_num, num_parts)
```
And the loading is as follows:
```
feat_path = os.path.join(os.path.dirname(part_config), 'wholegraph', ntype + '~feat')
node_feat_wm_embedding.get_embedding_tensor().from_file_prefix(feat_path,
                                                                       part_count=num_parts)
```

Reply:
Another thing is, when num_part changes, you don't need to re-run this conversion if using single binary file for wholegraph features 😄

Reply:
Thats true!!